include/drivers/ipm.h

```diff
@@ -20,9 +20,6 @@
  * @{
  */
 
-#include <kernel.h>
-#include <device.h>
```

Comment:
Why is that the right place? This header depends on device.h for `struct device *` in the callback typedefs.

Reply:
This depends on the way how we include headers. One way is to include all possible headers in every header, creating dependency loop resolved with ifdefs. 
Another way is to keep minimum needed headers. The files including this API header `ipm.h` should include `device.h` before this header. As you see from the patches in some cases `device.h` is already included, but in some cases not. Since we have those ifdefs, we can include headers multiple times.
However is this solution is not acceptable we can mention this and also make code consistent (include all possible header in API headers like this `ipm.h`).